integrationtest/mapper/orm-coordination-outbox-polling/src/test/java/org/hibernate/search/integrationtest/mapper/orm/coordination/outboxpolling/automaticindexing/OutboxPollingAutomaticIndexingOutOfOrderIdsIT.java

```diff
@@ -370,13 +371,41 @@ private void updateOutboxTableRow(Session session, Integer oldId, Integer newId)
 				assumeTrue(
 						"The H2 actual maximum available precision depends on operating system and JVM and can be 3 (milliseconds) or higher. " +
 								"Higher precision is not available before Java 9.",
-						!(oldJavaVersion && env.getDialect() instanceof H2Dialect)
+						!( oldJavaVersion && env.getDialect() instanceof H2Dialect )
 				);
 			}
 
-			try ( PreparedStatement ps = jdbc.getStatementPreparer().prepareStatement( OUTBOX_EVENT_UPDATE_ID ) ) {
-				ps.setInt( 1, newId );
-				ps.setInt( 2, oldId );
+			List<String> uuids = new ArrayList<>();
+			List<java.sql.Timestamp> times = new ArrayList<>();
+			try ( PreparedStatement statement = jdbc.getStatementPreparer().prepareStatement( OUTBOX_EVENT_SELECT_ORDERED_IDS_AND_CREATED_TIME ) ) {
+				ResultSet resultSet = statement.executeQuery();
+				while ( resultSet.next() ) {
+					uuids.add( resultSet.getString( 1 ) );
+					times.add( resultSet.getTimestamp( 2 ) );
+				}
+			}
+
+			String temporaryUuid = UUID.randomUUID().toString();
+			try ( PreparedStatement ps = jdbc.getStatementPreparer().prepareStatement( OUTBOX_EVENT_UPDATE_ID_AND_TIME ) ) {
+				ps.setString( 1, temporaryUuid );
+				ps.setTimestamp( 2, times.get( row1 ) );
+				ps.setString( 3, uuids.get( row2 ) );
+
+				jdbc.getResultSetReturn().executeUpdate( ps );
+			}
+
+			try ( PreparedStatement ps = jdbc.getStatementPreparer().prepareStatement( OUTBOX_EVENT_UPDATE_ID_AND_TIME ) ) {
+				ps.setString( 1, uuids.get( row2 ) );
+				ps.setTimestamp( 2, times.get( row2 ) );
+				ps.setString( 3, uuids.get( row1 ) );
+
+				jdbc.getResultSetReturn().executeUpdate( ps );
+			}
+
+			try ( PreparedStatement ps = jdbc.getStatementPreparer().prepareStatement( OUTBOX_EVENT_UPDATE_ID_AND_TIME ) ) {
+				ps.setString( 1, uuids.get( row1 ) );
+				ps.setTimestamp( 2, times.get( row1 ) );
+				ps.setString( 3, temporaryUuid );
 
 				jdbc.getResultSetReturn().executeUpdate( ps );
 			}
```

Comment:
Would you mind moving this repeated code to a method, say... `updateOutboxTableRow(Session session, UUID oldId, UUID newId, java.sql.Timestamp newCreated)`? :) It's a bit hard to understand as it is, even if I can guess what it's supposed to do.